app/models/manageiq/providers/openstack/cloud_manager/auth_key_pair.rb

```diff
@@ -1,2 +1,23 @@
 class ManageIQ::Providers::Openstack::CloudManager::AuthKeyPair < ManageIQ::Providers::CloudManager::AuthKeyPair
+
+  def self.create_keypair(ems, name, public_key)
```

Comment:
might be better to just accept create_options here? Seems better to me, since the same methods will be on other providers, which could have different params.